apis/quay/v1/quayregistry_types.go

```diff
@@ -88,6 +88,16 @@ type Component struct {
 	// Managed indicates whether or not the Operator is responsible for the lifecycle of this component.
 	// Default is true.
 	Managed bool `json:"managed"`
+	// Overrides holds information regarding component specific configurations.
+	Overrides Override `json:"overrides,omitempty"`
+}
+
+// Override describes configuration overrides for the given managed component
+type Override struct {
+	// Env holds environment variable overrides that will be added to the given managed components deployment (if available)
+	Env []string `json:"env,omitempty"`
```

Comment:
We could make this a struct so we limit what env vars can be exported to services.
Wdyt?

Reply:
Talked to @dmesser about this and for now it's better for us to stay flexible so we cover the "unknown unknowns".
With that said, shouldn't `Env` be a `map[string]string` ?